### DISCUSSION TITLE
Is it efficient to query data in Rust and send it to the frontend in Tauri, or should I use an SQL plugin for better performance?

### DISCUSSION THREAD

**Discussion by congchuahiep** [tauri-apps/tauri#11845] [Q&A] [answered]
I'm developing a Tauri application and currently handling data querying in two ways:

1. Querying data in Rust (using a MySQL/SQLite library) and then sending the result to the frontend via Tauri commands.

2. Using an SQL plugin on the frontend side to directly query the database.

I'm concerned about performance and efficiency. Which approach is better in terms of: Execution speed, Data transfer latency, Memory usage, Security

Would using an SQL plugin significantly improve performance, or is querying through Rust more optimal for large datasets and real-time scenarios? Additionally, what are the pros and cons of each approach?

**Comment by FabianLars** [comment, marked as answer]
Both options are essentially the same. The sql plugin is nothing special, it also just calls the rust side and there uses the sqlx plugin.
Using rust yourself directly is likely better as you can optimize the implementation to what you actually need, and not have to use the general purpose impl of the plugin. Also it would enable you to pre-process the responses on the rust side and sending only what you need to the frontend which would help with the big IPC bottleneck.
